uint16 opt_spot
string wpts